client/src/scenes/PuzzleGame.lua

```diff
@@ -60,37 +61,19 @@ function PuzzleGame:customRun()
   end
 end
 
-function PuzzleGame:runGameOver()
-  GraphicsUtil.print(self.text, (consts.CANVAS_WIDTH - GraphicsUtil.getGlobalFont():getWidth(self.text)) / 2, 10)
-
-  self.match:run()
-
-  -- if a key of the used inputconfig is used, set the player to want ready again
-  local keyPressed = tableUtils.trueForAny(self.inputConfiguration.isDown, function(key) return key end)
-
-  if (keyPressed) then
-    GAME.theme:playValidationSfx()
-    SFX_GameOver_Play = 0
-    if self.match.players[1].settings.puzzleIndex <= #self.match.players[1].settings.puzzleSet.puzzles then
-      self.match.players[1]:setWantsReady(true)
-    else
-      GAME.navigationStack:pop()
-    end
+function PuzzleGame:startNextScene()
+  if self.match.players[1].settings.puzzleIndex <= #self.match.players[1].settings.puzzleSet.puzzles then
+    self.match.players[1]:setWantsReady(true)
+  else
+    GAME.navigationStack:pop()
```

Comment:
This doesn't really work for puzzles.
Other game modes accept any key for leaving the game screen but in PuzzleMode, this key press also represents a Ready Up so it HAS to be a key pressed on at the very least an existing key config.
So far this has been achieved by storing the player config on the scene and only react to presses from that input configuration.
As it is, if you press any unbound key that has predefined meaning (e.g. Escape) after finishing a puzzle, the game bricks.